components/automate-deployment/pkg/manifest/client/http.go

```diff
@@ -150,10 +150,10 @@ func (c *HTTP) GetCurrentManifest(ctx context.Context, channel string) (*manifes
 	//try to get semantic version manifest
 	url := fmt.Sprintf(c.latestSemanticManifestURLFmt, channel)
 	m, err := c.manifestFromURL(ctx, url)
-	if err == nil {
+	if err == nil && m.SemVersion != "" && m.SchemaVersion == "2" {
 		return m, nil
 	}
-	if strings.Contains(err.Error(), "failed to locate manifest") {
+	if (m != nil && (m.SemVersion == "" || m.SchemaVersion == "1")) || strings.Contains(err.Error(), "failed to locate manifest") {
```

Comment:
Please verify this condition for error, as this is having OR condition.

Reply:
Here are the possible cases:

1. No Semantic Version or fetch semantic version returns error:
    1. M is null in this case, and err is not null - 
2. Semantic Version with incorrect schema version and empty version number (current scenario):
    1. M is not null and error will be null, and m.version is empty and m.schemaVersion is 1 
3. Semantic Version with correct results:
    1. This code won’t execute as err is nil and m.version is not empty and m.schemaVersion is 2

The above works for all above scenarios. Please suggest if I miss any other scenario.